extensions/cornerstone/src/hooks/useViewportActionCorners.ts

```diff
@@ -0,0 +1,123 @@
+import { useEffect } from 'react';
+import React from 'react';
+import { AllInOneMenu } from '@ohif/ui-next';
+import { getWindowLevelActionMenu } from '../components/WindowLevelActionMenu/getWindowLevelActionMenu';
+import { getViewportDataOverlaySettingsMenu } from '../components/ViewportDataOverlaySettingMenu';
+import { getViewportOrientationMenu } from '../components/ViewportOrientationMenu';
+
+/**
+ * Viewport action corner location type
+ */
+export interface ViewportActionCornerLocation {
+  type: 'VIEWPORT';
+  id: string;
+  corner?: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+}
+
+/**
+ * Viewport action corner service interface
+ */
+export interface ViewportActionCornerService {
+  addComponent: (params: {
+    viewportId: string;
+    id: string;
+    component: React.ReactNode;
+    location: ViewportActionCornerLocation | string;
+    indexPriority?: number;
+  }) => void;
+  clear: (viewportId: string) => void;
+  getAlignAndSide: (location: ViewportActionCornerLocation | string) => {
+    align: 'start' | 'center' | 'end';
+    side: 'top' | 'right' | 'bottom' | 'left';
+  };
+}
+
+/**
+ * Customization service interface
+ */
+export interface CustomizationService {
+  getCustomization: (name: string) => any;
+}
+
+/**
+ * Props for the useViewportActionCorners hook
+ */
+interface UseViewportActionCornersProps {
+  viewportId: string;
+  elementRef: React.MutableRefObject<HTMLDivElement>;
+  displaySets: AppTypes.DisplaySet[];
+  viewportActionCornersService: ViewportActionCornerService;
+  customizationService: CustomizationService;
+  commandsManager: AppTypes.CommandsManager;
+}
+
+/**
+ * Hook to manage viewport action corners for a Cornerstone viewport
+ */
+export function useViewportActionCorners({
+  viewportId,
+  elementRef,
+  displaySets,
+  viewportActionCornersService,
+  customizationService,
+  commandsManager,
+}: UseViewportActionCornersProps): void {
+  // Set up the window level action menu in the viewport action corners.
+  useEffect(() => {
+    const windowLevelActionMenu = customizationService.getCustomization(
```

Comment:
Can you change this code to just iterate over a set of named customizations, and add them one at a time to the viewport corners service?  I really dislike the knowledge of the ordering and construction of the child elements.  Just pass in the values as standard components.  I know we discussed making it a first class toolbar, which I'd really prefer, but at a minimum define a list of components and instantiate them generically.